plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceManualRangeWithIndicesCallsInspection.kt

```diff
@@ -135,26 +124,23 @@ class ReplaceIndexLoopWithCollectionLoopQuickFix : LocalQuickFix {
         val newReferenceExpression = factory.createExpression("element")
         arrayAccessElement.replace(newReferenceExpression)
         loopParameter.replace(newParameter)
-        loopRange.replace(collection)
+        loopRange.replace(collection ?: factory.createThisExpression())
     }
 }
 
 private fun KtExpression.toIntConstant(): Int? {
     return (this as? KtConstantExpression)?.text?.toIntOrNull()
 }
 
-fun KtExpression.receiverIfIsSizeOrLengthCall(): KtExpression? {
-    if (this.isSizeOrLength()) {
-        return (this as? KtDotQualifiedExpression)?.receiverExpression ?: return null
-    }
-    return null
-}
-
-fun KtExpression.receiverIfIsSizeOrLengthMinusOneCall(): KtExpression? {
-    if (this !is KtBinaryExpression) return null
-    if (this.operationToken != KtTokens.MINUS) return null
-    val collection = this.left?.receiverIfIsSizeOrLengthCall() ?: return null
-    val constant = this.right?.toIntConstant() ?: return null
-    if (constant == 1) return collection
-    return null
+private fun KtExpression.sizeOrLengthCall(rangeFunction: String): KtExpression? {
+    val expression = when(rangeFunction) {
+        "until" -> this
```

Comment:
the issue exists before this PR: `rangeFunction` is calculated as `rangeFunctions.find { it == binaryExpression.operationReference.text }` that is completely wrong  - it could be user-defined operator that is not resolved into e.g. `kotlin.ranges.rangeTo`.

The fix could be out scope of this PR, but it would be nice to fix it.